app/proc/telegram.go

```diff
@@ -48,18 +51,97 @@ func (client TelegramClient) Send(channelID string, item feed.Item) error {
 		return nil
 	}
 
+	contentLength, err := getContentLength(item.Enclosure.URL)
+	if err != nil {
+		return err
+	}
+
+	var message *tb.Message
+
+	if contentLength < 50_000_000 {
+		message, err = client.sendAudio(channelID, item)
+	} else {
+		message, err = client.sendText(channelID, item)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] send telegram message: \n%s", message.Text)
+	return err
+}
+
+func getContentLength(url string) (int64, error) {
+	resp, err := http.Head(url) //nolint:gosec
+	if err != nil {
+		return 0, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return 0, errors.New("status code != 200")
+	}
+
+	log.Printf("[DEBUG] Content-Length: %d, url: %s", resp.ContentLength, url)
+	return resp.ContentLength, err
+}
+
+func (client TelegramClient) sendText(channelID string, item feed.Item) (*tb.Message, error) {
 	message, err := client.Bot.Send(
 		recipient{chatID: channelID},
 		client.getMessageHTML(item),
 		tb.ModeHTML,
 		tb.NoPreview,
 	)
+
+	return message, err
+}
+
+func (client TelegramClient) sendAudio(channelID string, item feed.Item) (*tb.Message, error) {
+	file, err := client.downloadAudio(item.Enclosure.URL)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	log.Printf("[DEBUG] send telegram message: \n%s", message.Text)
-	return err
+	audio := tb.Audio{
+		File:     tb.FromReader(bytes.NewReader(*file)),
+		FileName: "1.mp3",
+		MIME:     "audio/mpeg",
+		Caption:  client.getMessageHTML(item),
+		Title:    item.Title,
+	}
+
+	message, err := audio.Send(
+		client.Bot,
+		recipient{chatID: channelID},
+		&tb.SendOptions{
+			ParseMode: tb.ModeHTML,
+		},
+	)
+
+	return message, err
+}
+
+func (client TelegramClient) downloadAudio(url string) (*[]byte, error) {
+	clientHTTP := &http.Client{Timeout: 60 * 10 * time.Second}
```

Comment:
I set big timeout, it is OK ?

Reply:
hard-coded timeout is something to avoid. But I can add it to opts and pass in latter on